src/Alpha_complex/include/gudhi/Incremental_delaunay.h

```diff
@@ -0,0 +1,137 @@
+#include <iterator>
+#include <vector>
+
+namespace Gudhi {
+
+/** \ingroup alpha_complex
+ *
+ * This function constructs the incremental Delaunay complex, as defined in \cite incremental-delaunay .
+ * Given a list of points (p1, ..., pn), the incremental Delaunay complex contains the Delaunay triangulation
+ * of (p1, ..., pk) for all k≤n. For any simplex s of the triangulation of (p1, ..., pk) that is not in the
+ * triangulation of (p1, ..., pn), the complex also contains a simplex formed with the same points as s plus the
+ * first pj that conflicts with s (is inside its circumsphere). For points in R^d, the complex usually has
+ * dimension d+1.
+ *
+ * This function calls `complex.insert_simplex_and_subfaces` on a set of simplices large enough that, with their
+ * faces, they define the incremental Delaunay complex. However, this set is not guaranteed to be minimal, there
+ * may be redundancies.
+ *
+ * No filtration values are computed here, those can be computed separately afterwards, for instance
+ * with `Gudhi::cech_complex::assign_MEB_filtration`.
+ *
+ * @tparam K CGAL kernel: either Epick_d or Epeck_d.
+ * @tparam SimplicialComplex Simplicial complex. It must provide a type `Vertex_handle` constructible from `int`
+ *   to name vertices, and a member function `void insert_simplex_and_subfaces(vrange)` which accepts a range
+ *   of `Vertex_handle` (representing a simplex).
+ * @tparam PointRange Forward range of `K::Point_d`.
+ *
+ * @param[in] k The geometric kernel.
+ * @param[out] complex The simplicial complex.
+ * @param[in] points Embedding of the points.
+ *
+ * Authors: Michael Kerber, Marc Glisse
+ */
+template<typename K, typename SimplicialComplex, typename PointRange>
+void construct_incremental_delaunay(K const&k, SimplicialComplex& complex, PointRange const& points) {
+  using TDS = CGAL::Triangulation_data_structure<
+    typename K::Dimension,
+    CGAL::Triangulation_vertex<K, typename SimplicialComplex::Vertex_handle>,
+    CGAL::Triangulation_ds_full_cell<void, CGAL::TDS_full_cell_mirror_storage_policy> >;
+  using Triangulation = CGAL::Delaunay_triangulation<K, TDS>;
+  auto pt_it = std::begin(points);
+  auto pt_end = std::end(points);
+  if(pt_it == pt_end) return;
+  int d = k.point_dimension_d_object()(*pt_it);
+  Triangulation tri(d);
+
+  // local variables pulled out for recycling
+  typedef std::vector<typename Triangulation::Full_cell_handle> Full_cell_h_vector;
+  Full_cell_h_vector cs;
+  std::vector<typename SimplicialComplex::Vertex_handle> splx;
+
+  for(int idx = 0; pt_it != pt_end; ++pt_it, ++idx) {
+    // Adapted from Delaunay_triangulation::insert(Point)
+    auto&& p = *pt_it;
+    typename Triangulation::Locate_type lt;
+    typename Triangulation::Face f(d);
+    typename Triangulation::Facet ft;
+    typename Triangulation::Full_cell_handle s = tri.locate(p, lt, f, ft);
+    // Adapted from Delaunay_triangulation::insert(Point,Locate_type,Face,Facet,Full_cell_handle)
```

Comment:
Should we mention it in the copyright ?

Reply:
Would
```
Authors: Michael Kerber, Marc Glisse, (Samuel Hornus for the original CGAL code)
```
in both places work?

Reply:
I am not sure (not a specialist), but keep it like that. I don't know if we should ask CGAL people if it fits their requirements, but you have more experience than me on this.